ckanext/xloader/utils.py

```diff
@@ -175,6 +178,68 @@ def get_xloader_user_apitoken():
     return site_user["apikey"]
 
 
+def _modify_url(input_url: str, base_url: str) -> str:
+    """ Modifies the input URL with base_url provided.
+
+    Args:
+        input_url (str): The original URL to potentially modify
+        base_url (str): The base URL to compare/replace against
+    Returns:
+        str: The modified URL with replaced scheme and netloc
+    """
+    parsed_input_url = urlparse(input_url)
+    parsed_base_url = urlparse(base_url)
+    # Do not modify non-HTTP(S) URLs (e.g., ftp://)
+    if parsed_input_url.scheme not in ("http", "https"):
+        return input_url
+    # replace scheme: "http/https" and netloc:"//<user>:<password>@<host>:<port>/<url-path>"
+    new_url = urlunparse(
+    (parsed_base_url.scheme,
+     parsed_base_url.netloc,
+     parsed_input_url.path,
+     parsed_input_url.params,
+     parsed_input_url.query,
+     parsed_input_url.fragment))
+    return new_url
+
+
+def modify_input_url(input_url: str) -> str:
+    """Returns a potentially modified CKAN URL.
+
+    This function takes a possible CKAN URL and potentially modifies its base URL while preserving the path,
+    query parameters, and fragments. The modification occurs only if three conditions are met:
+    1. The base URL of the input matches the configured CKAN site URL (ckan.site_url).
+    2. A `ckanext.xloader.site_url` is configured in the settings.
+    3. A `ckanext.xloader.site_url_ignore_path_regex` if configured in the settings and does not match.
+
+    Args:
+        input_url (str): The original CKAN URL to potentially modify
+    Returns:
+        str: Either the modified URL with new base URL from xloader_site_url,
+             or the original URL if conditions aren't met
+    """
+
+    xloader_site_url = config.get('ckanext.xloader.site_url')
+    if not xloader_site_url:
```

Comment:
This only makes sense if we don't configure a default value.

Reply:
default value is removed, so its None. 
Exit early no point wasting cpu cycles on the rest of the function.